Sources/ResilientDecoding/ResilientArray+DecodingOutcome.swift

```diff
@@ -0,0 +1,98 @@
+// Created by George Leontiev on 4/19/20.
+// Copyright © 2020 Airbnb Inc. All rights reserved.
+
+import Foundation
+
+extension Resilient {
+  
+  init<T>(_ results: [Result<T, Error>]) where Value == [T] {
+    self.init(results, transform: { $0 })
+  }
+  
+  init<T>(_ results: [Result<T, Error>]) where Value == [T]? {
+    self.init(results, transform: { $0 })
+  }
+
+  private init<T>(_ results: [Result<T, Error>], transform: ([T]) -> Value) {
+    let elements = results.compactMap { try? $0.get() }
+    let value = transform(elements)
+    if elements.count == results.count {
+      self.init(value, outcome: .decodedSuccessfully)
+    } else {
+      #if DEBUG
+      let error = ResilientDecodingOutcome.ArrayDecodingError(results: results)
+      /// `ArrayDecodingError` is not reported
+      self.init(value, outcome: .recoveredFrom(error, wasReported: false))
+      #else
+      self.init(value, outcome: .recoveredFromDebugOnlyError)
+      #endif
+    }
+  }
+
+}
+
+#if DEBUG
+
+extension ResilientDecodingOutcome {
+  
+  /**
+   A type representing some number of errors encountered while decoding an array
+   */
+  public struct ArrayDecodingError<Element>: Error {
+    public let results: [Result<Element, Error>]
+    public var errors: [Error] {
+      results.compactMap { result in
+        switch result {
+        case .success:
+          return nil
+        case .failure(let error):
+          return error
+        }
+      }
+    }
+    /// `ArrayDecodingError` should only be initialized in this file
+    fileprivate init(results: [Result<Element, Error>]) {
+      self.results = results
+    }
+  }
+  
+  fileprivate func arrayDecodingError<T>(_ elementType: T.Type = T.self) -> ResilientDecodingOutcome.ArrayDecodingError<T> {
```

Comment:
Why does `elementType` exist if it's never used?

Reply:
This actually _is_ vestigial.